Dao UI improvements

ManfredKarrer opened the pull request:
Too much small changed to list all....

ManfredKarrer committed:
Improve UI

ManfredKarrer committed:
Merge branch 'master' into DAO-UI-hide-details

ManfredKarrer committed:
Add full node mode and rpc data to settings

- Change FULL_DAO_NODE option from boolean to string to be able to detect not set values
- isFullDaoNode, rpcUser and rpcPw can be set in the UI. If prog ars are
available they will overwrite the UI data.

ManfredKarrer committed:
Update link

sqrrm commented on core/src/main/java/bisq/core/dao/DaoFacade.java:
```diff
@@ -358,16 +360,92 @@ public void publishBlindVote(Coin stake, ResultHandler resultHandler, ExceptionH
+            case PROPOSAL:
+                break;
+            case BREAK1:
+                firstBlock++;
```
This seems like the opposite of what the comment says. This will show the second block of the break as the first block.

ManfredKarrer replied:
Yes, should rename the var or refactor it so its less hacky...

sqrrm commented on core/src/main/java/bisq/core/dao/governance/proposal/compensation/CompensationProposalService.java:
```diff
@@ -62,11 +62,10 @@ public CompensationProposalService(BsqWalletService bsqWalletService,
             throws ValidationException, InsufficientMoneyException, TxException {
         this.requestedBsq = requestedBsq;
-        this.bsqAddress = bsqAddress;
+        this.bsqAddress = "B" + bsqWalletService.getUnusedAddress().toBase58();
```
Probably should have convenience functions toBSQAddress() and fromBSQAddress()

ripcurlx replied:
:+1:

ManfredKarrer replied:
Yes, good idea.

sqrrm commented on core/src/main/java/bisq/core/dao/state/governance/Param.java:
```diff
@@ -105,16 +105,21 @@
+    // block of the technical 3601 blocks is displayed as part of the break1 phase.
+  /*  PHASE_UNDEFINED(0),
+    PHASE_PROPOSAL(3601),      // 24 days
+    PHASE_BREAK1(149),        // 1 day
```
How about 299?

sqrrm reviewed (commented):
I'm generally ok with this. utACK after tending to the comments (especially on display of phase length)

ripcurlx commented on core/src/main/java/bisq/core/dao/DaoFacade.java:
```diff
@@ -358,16 +360,92 @@ public void publishBlindVote(Coin stake, ResultHandler resultHandler, ExceptionH
-
-    public int getDurationForPhase(DaoPhase.Phase phase) {
-        return periodService.getDurationForPhase(phase, daoStateService.getChainHeight());
+    // Because last block in request and voting phases must not be used fo making a tx as it will get confirmed in the
```
Typo:
> fo --> for

ripcurlx commented on core/src/main/java/bisq/core/dao/DaoFacade.java:
```diff
@@ -358,16 +360,92 @@ public void publishBlindVote(Coin stake, ResultHandler resultHandler, ExceptionH
+        return firstBlock;
+    }
+
+    // Because last block in request and voting phases must not be used fo making a tx as it will get confirmed in the
```
Typo:
> fo --> for

ripcurlx commented on core/src/main/java/bisq/core/dao/DaoFacade.java:
```diff
@@ -358,16 +360,92 @@ public void publishBlindVote(Coin stake, ResultHandler resultHandler, ExceptionH
+        return lastBlock;
+    }
+
+    // Because last block in request and voting phases must not be used fo making a tx as it will get confirmed in the
```
Typo:
> fo --> for

ripcurlx commented on core/src/main/resources/i18n/displayStrings_de.properties:
```diff
@@ -1291,7 +1291,7 @@ dao.wallet.menuItem.send=Senden
 dao.wallet.menuItem.transactions=Transaktionen
 
-dao.wallet.dashboard.statistics=Statistiken
+dao.wallet.dashboard.distribution=Statistiken
```
You don't need to bother about changing translations except displayStrings.properties. All other files will be overwritten by the Transifex state. So if we change keys that have been translated yet we need to copy it over in Transifex. Have to see how easy this is visible in the interface. I'll also have to walk over all supported translations with the redesign branch as lots of ":" in translations need to be removed.

ManfredKarrer replied:
Yes that was done by IDEA refacotring... Do u prefer if I revert such changes at non default property files the next time?

ripcurlx commented on desktop/src/main/java/bisq/desktop/main/dao/governance/proposals/ProposalsListItem.java:
```diff
@@ -101,33 +118,62 @@ public void cleanup() {
         if (phase == DaoPhase.Phase.PROPOSAL) {
-            icon = AwesomeDude.createIconLabel(AwesomeIcon.FILE_TEXT);
+            //icon = AwesomeDude.createIconLabel(AwesomeIcon.TRASH);
+            icon = AwesomeDude.createIconLabel(AwesomeIcon.TRASH);
```
Please use `Formbuilder.getIcon` so it is easier when we need to switch to the newest AwesomeDude library.

ManfredKarrer replied:
Ok, will do that in next PR.

ripcurlx commented on desktop/src/main/java/bisq/desktop/main/settings/preferences/PreferencesView.java:
```diff
@@ -695,12 +727,67 @@ private void activateDisplayPreferences() {
+        isDaoFullNodeCheckBox.setOnAction(e -> {
+            String key = "daoFullModeInfoShown";
+            if (isDaoFullNodeCheckBox.isSelected() && preferences.showAgain(key)) {
+                String url = "https://bisq.network/docs/dao-full-node";
```
Maybe we could think about moving all URLs into one place so they get easier to maintain when the structure in the docs is changed.

ManfredKarrer replied:
Yes good idea. I use forwarders so we are independent from real url.

ripcurlx reviewed (changes requested):
utACK - besides some minor changes

ripcurlx commented:
@ManfredKarrer Also please merge master into the PR so the travis build can run.

ManfredKarrer committed:
Merge branch 'master' into DAO-UI-hide-details

ManfredKarrer committed:
Add getUnusedBsqAddressAsString method

ManfredKarrer committed:
Use FormBuilder.getIcon instead of AwesomeDude.createIconLabel

ManfredKarrer committed:
FIx missing arguments in test

ManfredKarrer committed:
FIx missing arguments in test, Ignore test (JMockit issue)

@Christoph Can you have a look why JMockit is not working anymore
in that module? TradesChartsViewModel is null and creates a nullpointer.
Adding RunWith("JMockit") as it was earlier does not work with changed gradle file
with newer JMockit version.
